docs/firestore/android.md

```diff
@@ -1,6 +1,63 @@
 ---
-title: Android Setup
-description: Manually integrate Cloud Firestore into your Android application.
+title: Firestore Android Initialization
+description: Firestore initialization with Android.
 ---
 
 # Android Manual Linking
+
+> The following steps are only required if your environment does not have access to React Native
+> auto-linking.
+
+#### Add Firestore to Gradle Settings
+
+**`android/settings.gradle`**:
+
+```groovy
+include ':@react-native-firebase_firestore'
+project(':@react-native-firebase_firestore').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-firebase/firestore/android')
+```
+
+#### Add Firestore to App Gradle Dependencies
+
+**`android/app/build.gradle`**:
+
+```groovy
+// ..
+dependencies {
+  // ..
+  implementation project(':@react-native-firebase_firestore')
+}
+```
+
+#### Add Firestore to Main Android Application:
+
+**`android/app/src/main/java/{yourApp}/MainApplication.java`**:
+
+```java
+// ..
+import io.invertase.firebase.app.ReactNativeFirebaseAppPackage;
+
+  // ..
+  protected List<ReactPackage> getPackages() {
+    return Arrays.asList(
+      new MainReactPackage(),
+      new ReactNativeFirebaseFirestorePackage(),
+      // ..
+```
+
+#### Enable multidex
+
+Adding Firestore to your Android app requires [`multiDexEnabled` to be set to `true`](https://developer.android.com/studio/build/multidex) in `android/app/build.gradle`:
+
```

Comment:
There should probably be a note here about adding the androidx.multidex dependency and altering your MainApplication to extend MultidexApplication as well like these 3 lines https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh#L127

Reply:
Also, this isn't included when auto-linking. Do you think we should be adding it automatically for the user?

Reply:
That seems touchy, I believe the object hierarchy in someone's MainApplication is not deterministic. You could do a heuristic but you can't know for sure. Additionally some people may have decided to drop support for API<21? I dunno

Reply:
Yeah, my thinking is we just hook onto the dex error and output a custom error message.

Reply:
Added the missing multidex setup! 😄

Reply:
Awesome :-)